Breeze/src/Breeze.Wallet/ChainExtensions.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using NBitcoin;
+
+namespace Breeze.Wallet
+{
+    public static class ChainExtensions
+    {
+        /// <summary>
+        /// Determines whether the chain is downloaded and up to date.
+        /// </summary>
+        /// <param name="chain">The chain.</param>        
+        public static bool IsDownloaded(this ConcurrentChain chain)
+        {
+            return chain.Tip.Header.BlockTime.ToUnixTimeSeconds() > (DateTimeOffset.Now.ToUnixTimeSeconds() - TimeSpan.FromHours(1).TotalSeconds);
+        }
+
+        /// <summary>
+        /// Gets the type of the coin this chain relates to.
+        /// Obviously this method and how we figure out what coin we're on needs to be revisited.
+        /// </summary>
+        /// <param name="chain">The chain.</param>
+        /// <returns></returns>
+        /// <exception cref="System.Exception">No support for this coin.</exception>
+        public static CoinType GetCoinType(this ConcurrentChain chain)
```

Comment:
Interesting approach, I think we should consider the coin type a protocol parameter and add the field to the Network or Consensus (or even ConsensusParams) classes in NBitcoin or NStratis and set it when building the network (like with the address hash version and the wif version).

That's good for now though.